back/app/uploaders/base_image_uploader.rb

```
@@ -83,25 +84,72 @@ def resize_to_fill_using_convert(img, target_width, target_height, gravity = 'Ce
+          exiftool_stderr: stderr,
+          exiftool_exit_status: status.exitstatus,
+          exiftool_error_signal: status.termsig
+        }
+      )
+      raise "Image stripping failed for #{@file.path}: #{stderr}"
     end
   end
 
-  # Strip the image of any profiles, comments or these PNG chunks: bKGD,cHRM,EXIF,gAMA,
-  # iCCP,iTXt,sRGB,tEXt,zCCP,zTXt,date.
-  # (https://imagemagick.org/script/command-line-options.php#strip)
-  #
-  # It is a bit heavy-handed and removes a lot of things, including colour profiles which
-  # can have an effect on the image rendering. It also recompresses the image. Overall, it
-  # seems to remove vibrancy from the image.
-  def strip
-    manipulate! do |img|
-      img.strip
-      img
+  private
+
+  # Helper method to execute external commands safely and capture output/status,
+  # with error handling and reporting.
+  def execute_command(command, *args)
+    stdout_str, stderr_str, status = nil # Initialize for rescue blocks
+
+    begin
+      stdout_str, stderr_str, status = Open3.capture3(command, *args)
+    rescue Errno::ENOENT => e
+      ErrorReporter.report_msg(
+        'External command not found.',
+        extra: {
+          command_attempted: "#{command} #{args.join(' ')}",
+          error_message: e.message,
+          file_path: @file.path,
+          environment_path: ENV.fetch('PATH')
+        }
+      )
+      raise "Command not found: '#{command}'. Is it installed and in PATH?"
+    rescue StandardError => e
+      ErrorReporter.report_msg(
+        'An unexpected error occurred executing external command.',
+        extra: {
+          command_attempted: "#{command} #{args.join(' ')}",
+          error_message: e.message,
+          error_class: e.class.name,
+          file_path: @file.path,
+          stdout_captured: stdout_str,
+          stderr_captured: stderr_str,
+          backtrace: e.backtrace.first(10)
+        }
+      )
+      raise # Re-raise the original error
     end
```

I think this is going a step too far. I would just call `Open3.capture3` directly and, if an exception is raised, let it bubble up.